keras/engine/topology.py

```diff
@@ -465,6 +465,17 @@ def __call__(self, x, mask=None):
             x: can be a tensor or list/tuple of tensors.
             mask: tensor or list/tuple of tensors.
         '''
+        # place the op on the right device:
+        if type(x) in [list, tuple] and not hasattr(x[0], '_keras_on_device'):
```

Comment:
This is because how devices are managed in theano. In order for an op to be run on a particular device, majority of its inputs should be available on that device (the rest will be copied implicitly). So we have to make sure that all inputs to the op are transferred to the device specified by _DEVICE global var.